include/cxx.h

```diff
@@ -826,6 +826,93 @@ template <typename T>
 Box<T>::Box(uninit) noexcept {}
 #endif // CXXBRIDGE1_RUST_BOX
 
+#ifndef CXXBRIDGE1_RUST_OPTION
+template <typename T>
+class Option final {
+public:
+  Option() noexcept;
+  Option(Option&&) noexcept;
+  Option(T&&) noexcept;
```

Comment:
Do you want to support copy operations as well? If not, explicitly deleting them can provide a nicer compiler diagnostic:

```
  // Option is not copyable
  Option(const Option&) = delete;
  Option(Option&&) noexcept;
  Option(T&&) noexcept;
```

Or if you do, I would add them here:

```
  Option(const Option&) noexcept;
  Option(Option&&) noexcept;
  Option(const T&) noexcept;
  Option(T&&) noexcept;
```